dataflow/src/main/java/org/checkerframework/dataflow/expression/ThisReference.java

```diff
@@ -13,7 +13,8 @@ public ThisReference(TypeMirror type) {
 
   @Override
   public boolean equals(@Nullable Object obj) {
-    return obj instanceof ThisReference;
+    return obj instanceof ThisReference
+        && type.toString().equals(((ThisReference) obj).type.toString());
```

Comment:
It is inefficient to call toString just for the purpose of comparison.  Instead, use a comparison method in `javax.lang.model.util.Types`.

Also, if you redefine `equals`, please define `hashCode` as well.